dwitter/tests/templatetags/test_insert_magic_links.py

```diff
@@ -20,7 +21,7 @@ def test_insert_magic_links_replaces_user_with_valid_characters(self):
     def test_insert_magic_links_bypasses_user_with_invalid_characters(self):
         self.assertEqual(
             'u/a1$',
-            'u/a1$'
+            insert_magic_links('u/a1$')
```

Comment:
Yeah, but now a test fails... So I have to either switch it back, or introduce a new regex test to avoid inserting anchors into partially correct usernames. I have came up with such a regex, but users would have to insert a space after the username. So I thought about allowing dots and commas after the username as well, but I don't know if it's all too good to deal with this issue like that. I said this on discord as well, but noone replied :/

Reply:
I'm ok with `u/a1$` becoming `<a href='/u/a1'>u/a1</a>$` which I believe is how it works today?

You can change this test to `insert_magic_links_does_not_linkify_invalid_username_characters` or something similar and change it to the way it works now.

Reply:
Yeah, but it's kind of annoying, and doesn't feel right. If a username contains invalid characters, why match it? It's invalid!

I dunno... maybe I can find a better way of handling this.

Reply:
Well, with hashtags it's whitespace or nothing, and that's before the hashtag :/... And then with endings there's the whole ordeal of what to define as punctuation, and what's just an invalid character. But as I said, I need to have a think about this.